### PR TITLE
refactor: submission improvements

### PR DESCRIPTION
**Pull request by niall-shaw**
Refactors and improvements to submissions API

Mainly, refactor to the invitation / entity data types.

### CODE REVIEW

File: packages/backend/src/submission/service.ts
Comment (jakubkoci) [inline packages/backend/src/submission/service.ts:140]: Good check 👍

File: packages/backend/src/submission/service.ts
Comment (niall-shaw) [reply packages/backend/src/submission/service.ts:159]: But it would be a different submission, and different submission ID? Only `pending` state submissions are updated.

File: packages/backend/src/server.test.ts
Comment (niall-shaw) [reply packages/backend/src/server.test.ts:200]: > Also, I missed that we don't reuse generateNewInvitation in this test

That's intentional, so we can check the email sent contains the specific email address

Comment (jakubkoci) [reply packages/backend/src/server.test.ts:200]: You can pass it as a parameter ;)

File: packages/backend/src/submission/service.ts
Comment (jakubkoci) [reply packages/backend/src/submission/service.ts:159]: I think it would be the same. Admin opens the submission with ID 123 in state `pending` and while looking at it, before he approves it, the submitter sends another submission, then we do an update of ID 123, but the admin sees just the previous of submission with ID 123.

Comment (niall-shaw) [reply packages/backend/src/submission/service.ts:159]: Then perhaps we should implement some sort of webhook feature where we listen for changes on a particular submission while the admin is viewing it? Or we can verify the data is the same when the admin attempts to approve / decline, and if not, do not alert the admin that changes have been made since latest review.

Comment (niall-shaw) [reply packages/backend/src/submission/service.ts]: Removed ability to update a submission

File: packages/backend/src/submission/mongoRepository.ts
Comment (jakubkoci) [inline packages/backend/src/submission/mongoRepository.ts:65]: How/when is this function called? 🤔

File: packages/backend/src/submission/service.ts
Comment (jakubkoci) [inline packages/backend/src/submission/service.ts:138]: Do we have a test for the schema ID validation?

File: packages/backend/src/server.test.ts
Comment (jakubkoci) [inline packages/backend/src/server.test.ts:240]: I don't think we have to check all those attributes in this test at all.
```suggestion

```

File: packages/backend/src/server.test.ts
Comment (jakubkoci) [inline packages/backend/src/server.test.ts:220]: I don't think we have to check all those attributes in this test at all.
```suggestion

```

**Review by jakubkoci — COMMENTED** [review]
Are we sill using uuid for invitation links and do you consider it secure if we do?

File: packages/backend/src/submission/mongoRepository.ts
Comment (jakubkoci) [reply packages/backend/src/submission/mongoRepository.ts:65]: It seems we don't use it right? We can remove it if not.

Comment (niall-shaw) [reply packages/backend/src/submission/mongoRepository.ts:65]: It would be called when approving an submission. i.e. if the invitation associated with the submission has no entityId, then the invitation would be updated with the new entityId to ensure future submissions created with that invitationUrl correspond to the same entity.

**Comment by niall-shaw** [comment]
> Are we sill using uuid for invitation links and do you consider it secure if we do?

After some research, I have changed to use https://github.com/paralleldrive/cuid2
This library is considerably more secure (~millions of years to crack instead of ~30 mins for uuid)

File: packages/backend/src/submission/mongoRepository.ts
Comment (niall-shaw) [reply packages/backend/src/submission/mongoRepository.ts:65]: So should we remove or keep it in place for the future?

Comment (jakubkoci) [reply packages/backend/src/submission/mongoRepository.ts:65]: Remove :)

File: packages/backend/src/server.test.ts
Comment (jakubkoci) [inline packages/backend/src/server.test.ts:118]: Nice 👍

**Review by jakubkoci — COMMENTED** [review]
Good job :) Just remove the unused function. I would approve but then I would have to reapprove so I'm just adding it as a comment but the rest looks good 👍

**Review by jakubkoci — APPROVED** [review]

**Pull request merged**
